controllers/tempo/microservices_controller_test.go

```diff
@@ -93,9 +102,124 @@ func TestReconcile(t *testing.T) {
 		assert.NoError(t, err)
 		assert.NotEmpty(t, list.Items)
 	}
+
 	// test status
 	updatedTempo := v1alpha1.Microservices{}
 	err = k8sClient.Get(context.Background(), nsn, &updatedTempo)
 	require.NoError(t, err)
 	assert.Equal(t, "1.5.0", updatedTempo.Status.TempoVersion)
+
+	// test status condition
+	assert.Equal(t, []metav1.Condition{{
+		Type:               "Ready",
+		Status:             "True",
+		LastTransitionTime: updatedTempo.Status.Conditions[0].LastTransitionTime,
+		Reason:             "Ready",
+		Message:            "All components are operational",
+	}}, updatedTempo.Status.Conditions)
+	// make sure LastTransitionTime is recent
+	assert.InDelta(t, metav1.NewTime(time.Now()).Unix(), updatedTempo.Status.Conditions[0].LastTransitionTime.Unix(), 60)
+}
+
+func TestDegraded(t *testing.T) {
+	// First, create object storage secret and Tempo CR
+	nsn := types.NamespacedName{Name: "degraded-test", Namespace: "default"}
+	storageSecret := createSecret(t, nsn)
+	createTempoCR(t, nsn, storageSecret)
+
+	// Reconcile
+	reconciler := MicroservicesReconciler{
+		Client: k8sClient,
+		Scheme: testScheme,
+	}
+	req := ctrl.Request{
+		NamespacedName: nsn,
+	}
+	reconcileResult, err := reconciler.Reconcile(context.Background(), req)
+	require.NoError(t, err)
+	assert.Equal(t, false, reconcileResult.Requeue)
+
+	// Verify status conditions: Ready=true
+	updatedTempo1 := v1alpha1.Microservices{}
+	err = k8sClient.Get(context.Background(), nsn, &updatedTempo1)
+	require.NoError(t, err)
+	assert.Equal(t, []metav1.Condition{{
+		Type:               "Ready",
+		Status:             "True",
+		LastTransitionTime: updatedTempo1.Status.Conditions[0].LastTransitionTime,
+		Reason:             "Ready",
```

Comment:
May be we can use the conditionReasons constants defined in the `microservices_types.go` ?

Reply:
Good idea, I've updated the PR.